pkg/scheduler/internal/cache/comparer/cache_comparer.go

```diff
@@ -66,10 +68,12 @@ func (c *cacheComparer) Compare() error {
 	return nil
 }
 
-type compareStrategy struct {
+// CompareStrategy is an implementation of the Scheduler's cache compare strategy.
+type CompareStrategy struct {
```

Comment:
I don't see where this type is referenced outside of this package. Does it need to be exported?

I suspect this is public because you can't initialize a struct from another package with internal fields.

The comment on this also implies that there is an interface somewhere being satisfied, but I couldn't find one. It's a bit strange that this code was organized this way. Not something that needs to be addressed in this PR though.

Reply:
I think maybe it should. `CacheComparer` and its other fields are public.